clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -109,29 +109,71 @@
+            LeaderAndIsrRequest.schemaVersions(), LeaderAndIsrResponse.schemaVersions()),
+    STOP_REPLICA(5, "StopReplica", true,
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            StopReplicaRequest.schemaVersions(), StopReplicaResponse.schemaVersions()),
+    UPDATE_METADATA(6, "UpdateMetadata", true,
+            "Sent by the controller to other brokers in the cluster to get them to update their metadata about " +
+            "the cluster.",
+            UpdateMetadataRequest.schemaVersions(), UpdateMetadataResponse.schemaVersions()),
+    CONTROLLED_SHUTDOWN(7, "ControlledShutdown", true,
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            ControlledShutdownRequest.schemaVersions(), ControlledShutdownResponse.schemaVersions()),
+    OFFSET_COMMIT(8, "OffsetCommit",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            OffsetCommitRequest.schemaVersions(), OffsetCommitResponse.schemaVersions()),
+    OFFSET_FETCH(9, "OffsetFetch",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            OffsetFetchRequest.schemaVersions(), OffsetFetchResponse.schemaVersions()),
+    FIND_COORDINATOR(10, "FindCoordinator",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            FindCoordinatorRequest.schemaVersions(), FindCoordinatorResponse.schemaVersions()),
+    JOIN_GROUP(11, "JoinGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            JoinGroupRequest.schemaVersions(), JoinGroupResponse.schemaVersions()),
+    HEARTBEAT(12, "Heartbeat",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            HeartbeatRequest.schemaVersions(), HeartbeatResponse.schemaVersions()),
+    LEAVE_GROUP(13, "LeaveGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            LeaveGroupRequest.schemaVersions(), LeaveGroupResponse.schemaVersions()),
+    SYNC_GROUP(14, "SyncGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            SyncGroupRequest.schemaVersions(), SyncGroupResponse.schemaVersions()),
+    DESCRIBE_GROUPS(15, "DescribeGroups",
+            "Sent by an admin client to a broker to get detailed information (such as group membersip) about a specific consumer group.",
+            DescribeGroupsRequest.schemaVersions(), DescribeGroupsResponse.schemaVersions()),
+    LIST_GROUPS(16, "ListGroups",
+            "Sent by an admin client to a particular broker to get information about the consumers groups managed by that broker. " +
```

Comment:
Ditto as above.

Reply:
Yeah, just to explain why I'm using parens: From the PoV of the protocol there is no distinction between different kinds of client, so you could argue the docs shouldn't make a distinction. OTOH it's useful for the reader to know which messages apply only the consumers or producers etc.

Reply:
I totally agree with you even because using just the protocol, someone could implement a "new client" which is producer, consumer and admin at same time. Btw as you said it's useful to give the reader such information just distinguish between APIs which make sense for a producer, a consumer or an administrator.